### PR TITLE
Adds Syndicate Listening Post

### PR DESCRIPTION
**Pull request by DangerRevolution**
## About the PR

First-step of adding Syndicate Listening Post mid-round ghostrole
Left to do:

- [X] Add the shuttle itself
- [X] Syndicate Radio Operator - Ghost Role
- [X] Syndicate Guards - AI Antagonists (Thanks to @evilexecutive)
- [X] Syndicate AA Radio 
- [X] Repurpose code to make the radio station spawn mid-round
- [x] Localise ghost text for Listening Operator
- [X] Add Syndie Crew Monitor
- [x] Recreate the shuttle to fit new additions as mentioned above and Decorate!!

## Why / Balance

It was a fun ghostrole in Station 13, however being:
A) A ghostrole in Station 13
B) Positioned 4KM away, in a derelict outpost with nothing but cams and a comms server to look at; as well as your PDA was a bit mediocre.

This brings in a semi-mobile disguised operations base; included with a crew monitor and a fax machine, as well as two AI Guards.

This is; frankly, a RolePlay role. Especially with no guarantee there'll been be Syndies in the AO, it'll be up to the players onboard to guide what happens with the shuttle, and how much they interfere, and how much they monitor.

This also maps in the Syndie Fax as well as several stamps, plus some self-defence weapons to the shuttle.

It also adds something for Salvage / Security to seize.
## Technical details

Mostly a .yml addition with a rehash of LoneOP and Ninja code to let it spawn mid-round, as well as a new crew monitoring server to pick up suit sensors within 2KM (to be added).
## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase
I have added the first draft of the shuttle.
image

**Changelog**

:cl: DangerRevolution and VMSolidus
- add: Rumours say that Syndicate Agents have cracked encoded communications! Added the Syndicate Listening Post to the Syndicate's arsenal of espionage!

### CODE REVIEW

**Comment by VMSolidus** [comment]
Oh god you put the entities in wizden's namespace. Uhh I'll go ahead and move them to DeltaV namespace and make appropriate files.

**Comment by VMSolidus** [comment]
image
I'm going to fix a bunch of this too. We don't need to comment or modify wizden code.

**Comment by ps3moira** [comment]
> Oh god you put the entities in wizden's namespace. Uhh I'll go ahead and move them to DeltaV namespace and make appropriate files.

:trollface:

**Comment by DangerRevolution** [comment]
> image I'm going to fix a bunch of this too. We don't need to comment or modify wizden code.

I knew that I'd have to sort it out into DeltaV folders but it started off as trial and error and after the hours it took me to get it sorted I was too lazy to go back over it and move it into DeltaV folders. I appreciate it though :)
the comments are also leftover parts of experimenting with the existing code; that again, hours of tedium made me too lazy to take out. I can passover and move things tomorrow (afternoon GMT time).

**Comment by VMSolidus** [comment]
https://github.com/DeltaV-Station/Delta-v/assets/16548818/bcc39d67-6c45-4d74-a091-8910aa6c34a9

But that's okay because here's the radio guards, I'll make their spawner next and then commit! They have the same ammo capacity as a standard Viper, and infact they shoot the same bullets as a Viper. Sadly it's not possible to have them once every 5 seconds reload the entire magazine. But there's also a couple different EntityAmmoProviders, so I'm going to experiment with a few and see what options I like. 

As you can see, these assholes are insanely aggressive. And yes, he did infact pick up and fire an RPG at the end of that video. If they see a gun they like that is better than their stock gun, they WILL pick it up and use it.

**Comment by ps3moira** [comment]
> Recording.2023-12-07.234322.mp4 
> But that's okay because here's the radio guards, I'll make their spawner next and then commit! They have the same ammo capacity as a standard Viper, and infact they shoot the same bullets as a Viper. Sadly it's not possible to have them once every 5 seconds reload the entire magazine. But there's also a couple different EntityAmmoProviders, so I'm going to experiment with a few and see what options I like.

Can you make fire the gun at a slower rate? Also for custom Sprites for headset and PDA's there's already syndicate headsets and newly added Syndicate Agent PDA's.

**Comment by VMSolidus** [comment]
Yes, I can arbitrarily control their fire rate, and decrease it(I probably will, since maxing out the fire rate of a Viper is INSANE).

**Comment by DangerRevolution** [comment]
@evilexecutive 
When you commit I'd appreciate if you could take a quick walk around it? Just take the ghost role and look around. Syndie Cams and Crew Monitor are a *soon*, I think I've found how to make them read across grids.

**Comment by VMSolidus** [comment]
image
Not too bad, although I think we should definitely replace the standard reinforced walls with the same walls that the Nukie shuttle is made with. That'll visually read to salvagers who encounter it that "This is extremely fucking dangerous", especially when they hear the boots walking around inside it from the NPC guards that randomly wander its halls

**Comment by VMSolidus** [comment]
image
Alright while I was at it, I also took a crack at "Prettying up" the outpost. Adding its asteroid camouflage, replacing the NT standard reinforced walls with Syndicate standard Plastitanium. I replaced the window with syndicate standard plasma glass with blast doors(that can be shut by the radio operator on his bridge. I also added sleeping quarters for the outpost guards. My version of the map is saved as pirateradiosolidus.yml, I didn't overwrite yours because I figured you would probably want to take a look at my version and make your own changes before making a "Main" version of the map.

**Comment by VMSolidus** [comment]
image
Awesome, the map loads in correctly now, and the guard spawners WORK

**Comment by VMSolidus** [comment]
There's still more work to do on the map, but I'll go do that tomorrow after I've had some sleep. I need to make the map more practical for "Long Term Observation", so it needs an extremely minimalist Atmos, a Rec Room, Tool Room, and a Minimalist Botany. I already know where I'm going to put all 4 of those things.

**Comment by VMSolidus** [comment]
image
I may have accidentally finished the outpost (except for decoration stuff) instead of sleeping.

**Comment by DangerRevolution** [comment]
> image I may have accidentally finished the outpost (except for decoration stuff) instead of sleeping.

Drippyyy I can't wait to see this. Just woke up but I'll get back to you.

**Comment by DangerRevolution** [comment]
[8162819](https://github.com/DeltaV-Station/Delta-v/pull/526/commits/8162819e67cb1d0d5479633e6ba8506d474edab1)
I've actually managed to make the Syndie Crew Monitoring Server; it picks up Suit Sensors past the range of the *Crew* Monitoring Server. Cameras themselves are hardcoded to *not* broadcast past 200. Might just up the broadcast range on the Cameras themselves as this shouldn't impact... anything, the monitor and the server won't pick it up anyways as it's not in range.

BUT if it goes badly this could break cameras on every map...

**Comment by DangerRevolution** [comment]
Adding a camera monitor to the shuttle would require some creative thinking. normal cameras can only be read by the router if they're on the same grid, the monitor can only read the router on the same grid.
wireless camera routers don't have this issue; but they're also not very useful.

I've changed locales so far, added the crew monitor under "suspicious crew monitoring server", I'll map-in the changes and some extra additions + thrusters + I'll get an Atmos main to look at the map in about 6 hours from now.

**Comment by DangerRevolution** [comment]
Changed inheritance so now the Listener Ops inherit names.

All that's left really is to drip the shuttle out more and add thrusters; which I'll do within the 6 hours and have a look at letting any race spawn there. Only concern is how many minutes of functional Oxygen the shuttle has..

Down the line; Syndie Cam Monitor would probably require upstream changes to wired/wireless; more listening-op specific drip would be nice.

**Comment by VMSolidus** [comment]
According to my math, the listening outpost has enough air to last its two radio operators 39.3 hours.

**Comment by VMSolidus** [comment]
syndicatelisteningpost
The actual map for the Syndicate Listening Post is now done!

**Comment by DangerRevolution** [comment]
This PR is like; done done. We're happy with it's state and happy for it to be reviewed.

**Comment by DangerRevolution** [comment]
This PR is like; done done. We're happy with it's state and happy for it to be reviewed.

File: Resources/Locale/en-US/deltav/ghost/roles/ghost-role-component.ftl [hunk 1]
```diff
@@ -2,4 +2,7 @@ ghost-role-information-nukie-mouse-name = Nuclear Operative Mouse
 ghost-role-information-nukie-mouse-description = A Nuclear Operative reinforcement for the Syndicate.
 ghost-role-information-nukie-mouse-rules = Normal syndicate antagonist rules apply. Work with whoever called you in, and don't harm them.
                                     The crew is allowed to kill you without warning.
-                                    You are allowed to attack the crew and destroy the station without provocation.
```
Comment (DangerRevolution) [inline Resources/Locale/en-US/deltav/ghost/roles/ghost-role-component.ftl:5]: Can't figure out how to fix this >:(
the text is still there!!

Comment (VMSolidus) [reply Resources/Locale/en-US/deltav/ghost/roles/ghost-role-component.ftl:5]: I gotcha, the secret is you have to use a program other than notepad.

**Comment by IamVelcroboy** [comment]
Map checks out. I just have 1 request: Map some pajamas in. (Alternatively, we could add them to the SyndieVend but either way, they should have them here 🙂)

Other thing of note is that some of the internal airlocks are not "locked" variants. That's ok if intentional. Just wanted to point it out if you did not intend for that.

**Comment by DangerRevolution** [comment]
> Other thing of note is that some of the internal airlocks are not "locked" variants. That's ok if intentional. Just wanted to point it out if you did not intend for that. 

A) some of the non-essential locks are glass and unlocked due to give salvage a little leeway; the airlocks are ROBUST and it'll be tedious to break every single door down. The entrance locks are locked to give Radio Ops time to spring their trap, I'm open to @evilexecutive 's opinion though!

> Map checks out. I just have 1 request: Map some pajamas in. (Alternatively, we could add them to the SyndieVend but either way, they should have them here 🙂

B) The radio operators are dripped in the Pyjama's, so there's at least one pair guaranteed. I can add a spare in the bedrooms if you meant a *mapped* version of it?

**Comment by IamVelcroboy** [comment]
> A) some of the non-essential locks are glass and unlocked due to give salvage a little leeway; the airlocks are ROBUST and it'll be tedious to break every single door down. The entrance locks are locked to give Radio Ops time to spring their trap, I'm open to @evilexecutive 's opinion though!

Works for me! 

> B) The radio operators are dripped in the Pyjama's, so there's at least one pair guaranteed. I can add a spare in the bedrooms if you meant a _mapped_ version of it?

Radio operators are the ghost role? Not one of the AIs? 
If so, then all good. Don't really need extra then.

**Comment by VMSolidus** [comment]
> Map checks out. I just have 1 request: Map some pajamas in. (Alternatively, we could add them to the SyndieVend but either way, they should have them here 🙂)

image
So the radio operators spawn wearing comfortable Pyjamas, but it was an oversight that the guards have nothing comfortable to sleep in. I've given the guard sleeping quarters a bag for their PJs. 

image
I also reorganized the door locks and moved the door to the botany to instead come from the Rec Room. The only locked doors in the facility are now the "Secure" doors. The Entrance, Bridge, Armory, "Atmos", and power room are all locked doors now, but all other doors are no longer locked.

File: Resources/Prototypes/DeltaV/GameRules/events.yml [hunk 2]
```diff
@@ -45,3 +45,16 @@
...
+- type: entity # Delta-V : Midround Syndie Listening Station Spawn
+  id: PirateRadioSpawn
+  parent: BaseGameRule
+  noSpawn: true
+  components:
+  - type: StationEvent
+    earliestStart: 20
+    weight: 7.5 # Worth some thought as to how frequent this should be...
+    minimumPlayers: 25
+    reoccurrenceDelay: 180
```
Comment (Colin-Tel) [inline Resources/Prototypes/DeltaV/GameRules/events.yml]:
```suggestion
    maxOccurrences: 1
```
This should perform the function of only having it happen once per round.

Comment (DangerRevolution) [reply Resources/Prototypes/DeltaV/GameRules/events.yml]: thanks!
i've kept reocurrence delay in-case MaxOccurences fails

File: Resources/Prototypes/DeltaV/Entities/Clothing/Ears/headset_alt.yml
Comment (Colin-Tel) [inline Resources/Prototypes/DeltaV/Entities/Clothing/Ears/headset_alt.yml:1]: Why was this deleted?

File: Resources/Prototypes/DeltaV/GameRules/events.yml [hunk 3]
```diff
@@ -45,3 +45,17 @@
...
+  id: PirateRadioSpawn
+  parent: BaseGameRule
+  noSpawn: true
+  components:
+  - type: StationEvent
+    earliestStart: 20
+    weight: 7.5 # Worth some thought as to how frequent this should be...
+    minimumPlayers: 25
+    maxOccurences: 1
+    reoccurrenceDelay: 180 #fallback for maxoccurences failing
```
Comment (Colin-Tel) [inline Resources/Prototypes/DeltaV/GameRules/events.yml]: Why would maxOccurrences fail?

File: Resources/Prototypes/DeltaV/Entities/Clothing/Ears/headset_alt.yml
Comment (DangerRevolution) [reply Resources/Prototypes/DeltaV/Entities/Clothing/Ears/headset_alt.yml:1]: I don't know but the HoP headset within headset_alt.yml is contained within headset.yml; same folder just in a different file name

File: Resources/Prototypes/DeltaV/GameRules/events.yml (hunk 3 above)
Comment (DangerRevolution) [reply Resources/Prototypes/DeltaV/GameRules/events.yml]: I don't know but I like to be safe, I can remove it if necessary!

File: Resources/Locale/en-US/deltav/ghost/roles/ghost-role-component.ftl [hunk 4]
```diff
@@ -1,5 +1,8 @@
...
-                                    The crew is allowed to kill you without warning.
-                                    You are allowed to attack the crew and destroy the station without provocation. 
\ No newline at end of file
+                                           The crew is allowed to kill you without warning.
+                                           You are allowed to attack the crew and destroy the station without provocation. 					
+ghost-role-information-listeningop-name = Listening Post Operative
+ghost-role-information-listeningop-description = You are a Listening Post operative. Get into range, observe the station, intercept communications and assist any operatives in the area!
+ghost-role-information-listeningop-rules =	You are a Syndicate Operative tasked with the continous reporting and monitoring of the station, it's activities and of assisting any fellow operatives who may be aboard the station. As an antagonist, do whatever is required to complete this task.  Do NOT abandon your station, and make sure it does not fall into enemy hands! Hide your existence at any cost!
```
Comment (Colin-Tel) [inline Resources/Locale/en-US/deltav/ghost/roles/ghost-role-component.ftl]:
```suggestion
ghost-role-information-listeningop-rules =	You are a Syndicate Operative tasked with the continuous reporting and monitoring of the station and its activities, as well as assisting any fellow operatives who may be aboard the station. As an antagonist, do whatever is required for you to complete this task. Make sure your station doesn't fall into enemy hands and DO NOT abandon your station! Hide your existence at any cost!
```

File: Resources/Prototypes/DeltaV/Entities/Clothing/Ears/headset_alt.yml
Comment (Colin-Tel) [reply Resources/Prototypes/DeltaV/Entities/Clothing/Ears/headset_alt.yml:1]: Oh I didn't even see you moved it. Why didn't you just say you moved it lol

Comment (DangerRevolution) [reply Resources/Prototypes/DeltaV/Entities/Clothing/Ears/headset_alt.yml:1]: i didn't know :shrug: :smile:

**Comment by DangerRevolution** [comment]
Closed to make use of uh, branches; reopened as #578
